terraform/main.tf

```diff
@@ -97,6 +99,8 @@ locals {
       osdisk_size_controller  = "500"
       num_builders_x86        = 1
       num_builders_aarch64    = 1
+      binary_cache_public_key = "ghaf-infra-dev:EdgcUJsErufZitluMOYmoJDMQE+HFyveI/D270Cr84I="
```

Comment:
Also not blocking anything now, but an improvement idea: support setting multiple public keys here and properly parsing such values in `terraform/modules/azurerm-nix-vm-image/nix-eval.sh`. As it is now, setting more than one value here (space separated) seems to blow-up the parsing in `terraform/modules/azurerm-nix-vm-image/nix-eval.sh`.

Reply:
Yes, parsing more structured data to Nix is painful due to the layer of bash in between, and already was very painful.

This definitely could be improved. I initially did not add it for simplicity, but we could have `nix-eval.sh` emit some more structured data (JSON), and then pass it to nix (some encoding over CLI args, or have Nix read it).

I'd like to keep this simple for now - do we really need multiple pubkeys configured on a per-environment level?

Reply:
Yes, we can keep it simple now and later think if we need to support setting multiple keys etc.

Reply:
> I'd like to keep this simple for now - do we really need multiple pubkeys configured on a per-environment level?

After thinking a bit more about this, If we have some other set of "well-known binary caches" we want to have control over enabling them or not, I'd probably rather see this being a "boolean" flag in the `custom-nixos.nix` file that enables some more options if it's set. And potentially the more detailed configuration of that living in a module, so it's better testable by CI, which executes none of this.